lib/logflare_web/controllers/plugs/compressed_body_reader.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Plug.Parsers.CompressedBodyReader do
+  def read_body(conn, opts \\ []) do
+    content_encoding = Plug.Conn.get_req_header(conn, "content-encoding")
+
+    with {:ok, body, conn} <- Plug.Conn.read_body(conn, opts),
+         do: {:ok, try_decompress(body, content_encoding), conn}
```

Comment:
Ok, that can cause memory exhaustion on the node, but I will prepare the fix for that.